gloo.gl.tests.test_functionality._draw1 fails

Tested on Ubuntu 14.04 and Intel GPU (the test passes on the NVIDIA GPU).

```
======================================================================
FAIL: Test pyopengl GL backend for full functionality.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/cyrille/anaconda/envs/py27/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/data/git/vispy/vispy/testing/_testing.py", line 180, in skipper
    return f(*args, **kwargs)
  File "/home/cyrille/anaconda/envs/py27/lib/python2.7/site-packages/numpy/testing/decorators.py", line 146, in skipper_func
    return f(*args, **kwargs)
  File "/data/git/vispy/vispy/gloo/gl/tests/test_functionality.py", line 59, in test_functionality_pyopengl
    _test_functonality('pyopengl')
  File "/data/git/vispy/vispy/gloo/gl/tests/test_functionality.py", line 96, in _test_functonality
    _draw1()
  File "/data/git/vispy/vispy/gloo/gl/tests/test_functionality.py", line 507, in _draw1
    _check_result()
  File "/data/git/vispy/vispy/gloo/gl/tests/test_functionality.py", line 547, in _check_result
    assert_equal(pix1, (0, 0, 0))
AssertionError: Tuples differ: (32, 32, 32) != (0, 0, 0)

First differing element 0:
32
0

- (32, 32, 32)
+ (0, 0, 0)
```

--- a/vispy/gloo/gl/tests/test_functionality.py
+++ b/vispy/gloo/gl/tests/test_functionality.py
@@ -36,21 +36,21 @@ def teardown_module():
 @requires_application()
 def test_functionality_desktop():
     """ Test desktop GL backend for full functionality. """
-    _test_functonality('gl2')
+    _test_functionality('gl2')
 
 
 @requires_application()
 def test_functionality_proxy():
     """ Test GL proxy class for full functionality. """
     # By using debug mode, we are using the proxy class
-    _test_functonality('gl2 debug')
+    _test_functionality('gl2 debug')
 
 
 @requires_application()
 @requires_pyopengl()
 def test_functionality_pyopengl():
     """ Test pyopengl GL backend for full functionality. """
-    _test_functonality('pyopengl2')
+    _test_functionality('pyopengl2')
 
 
 @requires_application()
@@ -60,7 +60,7 @@ def test_functionality_es2():
         raise SkipTest('Skip es2 functionality test for now.')
     if not sys.platform.startswith('win'):
         raise SkipTest('Can only test es2 functionality on Windows.')
-    _test_functonality('es2')
+    _test_functionality('es2')
 
 
 def _clear_screen():
@@ -68,7 +68,7 @@ def _clear_screen():
     gl.glFinish()
 
 
-def _test_functonality(backend):
+def _test_functionality(backend):
     """ Create app and canvas so we have a context. Then run tests.
     """
     # use the backend
@@ -261,7 +261,7 @@ def _prepare_vis():
     
     # --- get information on attributes and uniforms
     
-    # Count attribbutes and uniforms
+    # Count attributes and uniforms
     natt = gl.glGetProgramParameter(hprog, gl.GL_ACTIVE_ATTRIBUTES)
     nuni = gl.glGetProgramParameter(hprog, gl.GL_ACTIVE_UNIFORMS)
     assert_equal(natt, 4)

--- a/vispy/gloo/gl/tests/test_functionality.py
+++ b/vispy/gloo/gl/tests/test_functionality.py
@@ -234,6 +234,15 @@ def _prepare_vis():
     # touch glDetachShader
     gl.glDetachShader(hprog, hvert)
     gl.glAttachShader(hprog, hvert)
+
+    # Bind all attributes - we could let this occur automatically, but some
+    # implementations bind an attribute to index 0, which has the unfortunate
+    # property of being unable to be modified.
+    gl.glBindAttribLocation(hprog, 1, 'a_1')
+    gl.glBindAttribLocation(hprog, 2, 'a_2')
+    gl.glBindAttribLocation(hprog, 3, 'a_3')
+    gl.glBindAttribLocation(hprog, 4, 'a_4')
+
     gl.glLinkProgram(hprog)
     
     # Test that indeed these shaders are attached
@@ -248,9 +257,6 @@ def _prepare_vis():
     
     # Use it!
     gl.glUseProgram(hprog)
-    
-    # Bind one attribute
-    gl.glBindAttribLocation(hprog, 1, 'a_2')
     
     # Check if all is ok
     assert_equal(gl.glGetError(), 0)

